hugo/content/docs/recipes/validation/dependency-loops.md

```diff
@@ -0,0 +1,258 @@
+---
+title: "Dependency loops"
+weight: 100
+---
+
+## What is the problem?
+
+If you are building composite data structures or have some computation flow, you might be interested whether the product, that you are generating, does contain any loops back to the already used product.
+
+For data structures you could think of a structure in C that references itself (without using the pointer notation). This would lead to an infinitely expanding data type, which is practically not doable.
+
+Or for control flows these loops can be interpreted as recursion detection, a function that calls itself.
+
+Regardless of what usecase you have, you might have an interest to detect those loops.
+
+## How to solve it?
+
+There are two approaches for a loop detection depending on the nature of your situation.
+
+### Simple nature
+
+#### Simple detection
+
+If you have a `1:n` relationship like the `super class`-to-`sub class` relation for classes, you can do it by simply walking along the parent route (or in this specific example the `super class`-route). Just keep in mind all visited places and if one parent is already in that list, you have detected a loop!
+
+#### Simple resolution
+
+Assuming that you have no loops back, you can resolve a list of dependencies.
+You do a simple depth-first-search, starting with the parent visiting the children afterwards (recursively).
+
+### Complex nature
+
+#### Complex detection
+
+If you have a `n:m` relationship like it is given for function calls (a function can be called by `m` function and can call `n` functions), you can solve the question for loops by creating a directed graph.
+
+In the example of function calls you have to store edges for each call from function A to every function B. The nodes are the set of all functions. The key algorithm is the search for the so-called strongly-connected components in the resulting graph.
+
+It is recommended not to implement your own version of that algorithm. Please use an existing solution! The algorithm is able to output every loop with all its members of that loop. Perfect for adding a custom validation!
+
+#### Complex resolution
```

Comment:
During the first reading, I didn't got, that this is a follow-up step after ensuring, that there are no cycles: Maybe, it was only me, maybe, there is some motivation missing at the beginning?